src/main/scala/com/itechart/ny_accidents/service/WeatherMappingService.scala

```diff
@@ -1,52 +1,80 @@
 package com.itechart.ny_accidents.service
 
 import com.google.inject.{Inject, Singleton}
-import com.itechart.ny_accidents.constants.GeneralConstants
-import com.itechart.ny_accidents.constants.GeneralConstants.{FIRST_STATION_ID, LAST_STATION_ID}
+import com.itechart.ny_accidents.constants.GeneralConstants.{FIRST_STATION_ID, HASH_DIFFERENCE, LAST_STATION_ID}
 import com.itechart.ny_accidents.database.NYDataDatabase
 import com.itechart.ny_accidents.database.dao.WeatherDAO
-import com.itechart.ny_accidents.entity.{WeatherEntity, WeatherForAccident}
+import com.itechart.ny_accidents.entity.{WeatherEntity, WeatherForAccident, WeatherStation}
 import com.itechart.ny_accidents.parse.WeatherParser
+import com.itechart.ny_accidents.spark.Spark
+import com.itechart.ny_accidents.spark.Spark.sparkSql.implicits._
 import com.itechart.ny_accidents.utils.{DateUtils, PostgisUtils}
+import org.apache.spark.sql.Dataset
 
+import scala.annotation.tailrec
 import scala.concurrent.Await
 import scala.concurrent.duration.Duration
 import scala.util.Try
 
 @Singleton
 class WeatherMappingService @Inject()(weatherDAO: WeatherDAO, weatherParser: WeatherParser) {
 
-  private val allStations = Await.result(NYDataDatabase.database.run(weatherDAO.allStations()), Duration.Inf)
+
+  var counter = 1
```

Comment:
Could you justify the need for a variable in singleton class?
How that's going to work in different threads?

Reply:
this variable is never used, removed it
https://github.com/itechartbigdatalab/NY-Accidents-Rate/pull/56/commits/0ca555bd973fed5eb36e11b9215abe2ce98024e1